### PR TITLE
Fix deprecated prop-types

### PR DESCRIPTION
**Pull request by fubz**
The module is now ready for React 16

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-number-editor",
-  "version": "4.0.2",
+  "version": "4.0.3",
```
Comment (tleunen) [inline package.json]: Could you revert this line please? :)
A new version will be set when releasing the package.

File: package.json [hunk 2]
```diff
@@ -44,15 +45,15 @@
...
+    "react": "^15.6.0",
+    "react-dom": "^15.6.0",
     "style-loader": "^0.13.1",
     "webpack": "^1.14.0",
     "webpack-dev-server": "^1.16.2"
   },
   "peerDependencies": {
-    "react": "0.14.x || ^15.0.0-rc",
-    "react-dom": "0.14.x || ^15.0.0-rc"
+    "react": "0.14.x || ^16",
```
Comment (tleunen) [inline package.json]: Would be best to keep v15 as well, while adding 16

**Comment by fubz** [comment]
Happy to help!  Thanks for the review 👍

File: package.json [hunk 3]
```diff
@@ -52,8 +52,8 @@
     "webpack-dev-server": "^1.16.2"
   },
   "peerDependencies": {
-    "react": "0.14.x || ^16",
-    "react-dom": "0.14.x || ^16"
+    "react": "0.14.x - ^16.x",
```
[the remark this answers is not in the document]
Comment (fubz) [reply package.json]: Turns out the carat doesn't work with ranges, if I remove the caret it will work as well.

https://i.imgur.com/74dVNrn.png

**Pull request merged**